src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/CounterGroup.cs

```diff
@@ -203,7 +203,7 @@ private void ResetCounters()
                     }
                     else if (counter is IncrementingPollingCounter ipCounter)
                     {
-                        ipCounter.UpdateMetric();
```

Comment:
We also have the same issue for IncrementingEventCounter on line 202, right?

Reply:
```IncrementingEventCounter.UpdateMetric``` call looks safe for calling under ```EventListener.EventListenersLock``` since it just sets ```_prevIncrement``` value
But ```IncrementingPollingCounter``` call ```_totalValueProvider``` action (an arbitrary action passed via ```IncrementingPollingCounter``` ctor) which can be dangerous for us

We can unify the reset logic for all ```IncrementingEventCounter```, ```IncrementingPollingCounter```, ```EventCounter``` to not update metrics under ```EventListener.EventListenersLock```